t.in.eoarchive.py

```diff
@@ -0,0 +1,615 @@
+    else:
+        epsg_arg = {'srid': "EPSG:{}".format(epsg)}
+    # create temp location from input without import
+    grass.verbose(_(f"Creating temporary location with EPSG:{epsg}..."))
+    grass.run_command('g.proj', flags='c', location=TMPLOC, quiet=True,
+                      **epsg_arg)
+
+    # switch to temp location
+    os.environ['GISRC'] = str(SRCGISRC)
+    proj = grass.parse_command('g.proj', flags='g')
+    if 'epsg' in proj:
+        new_epsg = proj['epsg']
+    else:
+        new_epsg = proj['srid'].split('EPSG:')[1]
+    if new_epsg != str(epsg):
+        grass.fatal(_("Creation of temporary location failed!"))
+
+    # return SRCGISRC, TMPLOC
+
+
+def get_actual_location():
+    """ Function that returns location and mapset of the current grass GISENV
+    Returns:
+        tgtloc (string): current location name
+        tgtmapset (string): current mapset name
+
+    """
+    global TGTGISRC, GISDBASE
+    # get actual location, mapset, ...
+    grassenv = grass.gisenv()
+    tgtloc = grassenv['LOCATION_NAME']
+    tgtmapset = grassenv['MAPSET']
+    GISDBASE = grassenv['GISDBASE']
+    TGTGISRC = os.environ['GISRC']
+    return tgtloc, tgtmapset
+
+
+def get_utmcells():
+    """ Function that queries the UTM Military Grid Reference System for Grid
+        tiles that overlap with the current computational region
+    Returns:
+        utm_tiles (list): list of UTM tiles that overlap with the current
+                          computational region
+
+    """
+    # save region
+    tmpregionname = 'tmp_region_utmgrid_' + str(os.getpid())
+    grass.run_command('v.in.region', output=tmpregionname, quiet=True)
+    rm_vectors.append(tmpregionname)
+
+    # get actual location, mapset, ...
+    tgtloc, tgtmapset = get_actual_location()
+
+    # create temporary location with epsg:4326
+    epsg = 4326
+    createTMPlocation(epsg)
+
+    # import grid
+    grass.run_command(
+        'v.proj', location=tgtloc, mapset=tgtmapset, input=tmpregionname,
+        output=tmpregionname, quiet=True)
+    grass.run_command('g.region', vector=tmpregionname, quiet=True)
+    grass.run_command(
+        'v.in.wfs',
+        url=options['wfs_mgrs'],
+        output='tmp_s2tiles_area', srs=epsg, name=options['wfs_name'],
+        flags='r', quiet=True)
+    utm_tiles = [x for x in grass.parse_command(
+        'v.db.select', map='tmp_s2tiles_area', column='name', flags='c')]
+
+    # switch to target location
+    os.environ['GISRC'] = str(TGTGISRC)
+
+    return utm_tiles
+
+
+def check_start_end(start, end):
+    """ Checks the validity of the user defined start and end date
+    Args:
+        start(string): user-defined start date in format YYYY-MM-DD
+        end(string): user-defined end date in format YYYY-MM-DD
+    Returns:
+        start_date(datetime): user-defined start date as datetime object
+        end_date(datetime): user-defined end date as datetime object
+
+    """
+    # set end if end is today
+    if end == "today":
```

Comment:
Does the date-time term "today" conform to any standard? I could not find it in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) and also not in the [Extended Date/Time Format (EDTF) Specification](https://www.loc.gov/standards/datetime/). According to EDTF, ".." can be used to indicate an open part (start or end) of an interval.

Reply:
No, it is not conform to a standard, it just used in similar mundialis modules. But you are right that this may be confusing. It is now adapted such that neither `start` or `end` are required. If `start` is omitted, the earliest possible date is assumes as `start`. If `end` is omitted, today's date is assumed. The general assumption being that if the user does not specify a date, the whole catalogue should be searched